### DISCUSSION TITLE
Does many classes affect SEO?

### DISCUSSION THREAD

**Discussion by karaka200** [tailwindlabs/tailwindcss#5682] [Help] [answered]
Given that we have to put many classes on the page to use TailwindCSS, does this have an effect on SEO and Site SPEED? Does the [GTmetrix](https://gtmetrix.com/) site give error to this number of classes?

Like this (this is a small part of the whole code of a page):

```
<div class="relative mb-6">
    <input type="text" name="mobile" id="inputMobile"
   class="peer placeholder-transparent w-full h-12 text-sm text-gray-900
   px-4 border border-gray-300 rounded-md transition-all shadow-sm focus:outline-none focus:border-blue-500
   focus:ring-1 focus:ring-blue-500"
   placeholder="Mobile" required>
    <label for="inputMobile" class="bg-white absolute -top-2.5 right-4 text-xs text-gray-400 px-2
    transition-all
    peer-placeholder-shown:bg-transparent peer-placeholder-shown:px-0
    peer-placeholder-shown:top-3.5 peer-placeholder-shown:text-sm peer-placeholder-shown:text-gray-500
    peer-focus:bg-white
    peer-focus:-top-2.5 peer-focus:right-4 peer-focus:text-xs peer-focus:text-blue-500 peer-focus:px-2">Mobile</label>
</div>

<div class="relative mb-6">
    <input type="password" name="password" id="inputPassword" class="peer placeholder-transparent w-full h-12 text-sm text-gray-900
   px-4 border border-gray-300 rounded-md transition-all shadow-sm focus:outline-none focus:border-blue-500
   focus:ring-1 focus:ring-blue-500" placeholder="password"
   autocomplete="off"
   required>
    <label for="inputPassword" class="bg-white absolute -top-2.5 right-4 text-xs text-gray-400 px-2
    transition-all
    peer-placeholder-shown:bg-transparent peer-placeholder-shown:px-0
    peer-placeholder-shown:top-3.5 peer-placeholder-shown:text-sm peer-placeholder-shown:text-gray-500
    peer-focus:bg-white
    peer-focus:-top-2.5 peer-focus:right-4 peer-focus:text-xs peer-focus:text-blue-500 peer-focus:px-2">Password</label>
    <span>Show</span>
</div>
```

**Comment by simonswiss** [comment, marked as answer]
Hey!

Adding multiple classes to your HTML elements doesn't affect SEO at all – what matters is you use the appropriate HTML tags to represent the content of your website 👍

As per page speed – while the HTML file size _can_ get bigger with all these classes, the issue becomes totally irrelevant when the files are gzipped, where the difference is very minimal.

Nicolas Gallagher did an interesting experiment on that exact thing, in this brilliant article he wrote in 2012 (and which is still extremely relevant):

http://nicolasgallagher.com/about-html-semantics-front-end-architecture/

The second to last section, called **A note on raw file size and HTTP compression**, is what I'm talking about.

Taking into consideration that utility classes are optimized for performance (compressibility in both the HTML and CSS files, shallow selectors, ..), turns out that utility CSS can be arguably _better_ for performance than some other alternative approaches.

Hope it helps!
